techlibs/ice40/synth_ice40.cc

```diff
@@ -311,7 +331,10 @@ struct SynthIce40Pass : public ScriptPass
 				run("techmap -map +/gate2lut.v -D LUT_WIDTH=4", "(only if -noabc)");
 			}
 			if (!noabc) {
-				run("abc -dress -lut 4", "(skip if -noabc)");
+				if (abc == "abc9")
+					run(abc + stringf(" -dress -lut +/ice40/abc_%s.lut -box +/ice40/abc_%s.box", device_opt.c_str(), device_opt.c_str()), "(skip if -noabc)");
```

Comment:
`-dress` has been removed from `abc9` now, so this is failing. Might also be worth adding a wire delay here (subject to experimentation)

Reply:
Good spot!